cmd/wigMath/wigMath.go

```diff
@@ -0,0 +1,197 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/wig"
+	"log"
+	"math"
+	"math/rand"
+)
+
+func wigMath(s Settings) {
+	rand.Seed(s.SetSeed)
+	var err error
+	var second []wig.Wig
+	var i, j, chromIndex int
+	records := wig.Read(s.InFile)
+	multipleOptionCheck(s)
+	if s.ScalarMultiply != 1 {
+		for i = range records {
+			for j = range records[i].Values {
+				records[i].Values[j] *= s.ScalarMultiply
+			}
+		}
+		wig.Write(s.OutFile, records)
+	} else if s.ScalarDivide != 1 {
+		if s.ScalarDivide == 0 {
+			log.Fatalf("You cannot divide wig values by zero, please input another value. I'm not mad but I'm a bit disappointed.")
+		}
+		for i = range records {
+			for j = range records[i].Values {
+				records[i].Values[j] /= s.ScalarDivide
+			}
+		}
+		wig.Write(s.OutFile, records)
+	} else if s.ElementWiseAdd != "" {
+		second = wig.Read(s.ElementWiseAdd)
+		for i = range records {
+			chromIndex = getChromIndex(second, records[i].Chrom)
+			for j = range records[i].Values {
+				records[i].Values[j] += second[chromIndex].Values[j]
+			}
+		}
+		wig.Write(s.OutFile, records)
+	} else if s.ElementWiseSubtract != "" {
+		second = wig.Read(s.ElementWiseSubtract)
+		for i = range records {
+			chromIndex = getChromIndex(second, records[i].Chrom)
+			for j = range records[i].Values {
+				records[i].Values[j] -= second[chromIndex].Values[j]
+			}
+		}
+		wig.Write(s.OutFile, records)
+	} else if s.MovingAverageSmoothing > 1 {
+		records = wig.SmoothSlice(records, s.MovingAverageSmoothing)
+		wig.Write(s.OutFile, records)
+	} else if s.AbsoluteError != "" {
+		second = wig.Read(s.AbsoluteError)
+		for i = range records {
+			chromIndex = getChromIndex(second, records[i].Chrom)
+			for j = range records[i].Values {
+				records[i].Values[j] = math.Abs(records[i].Values[j] - second[chromIndex].Values[j])
+			}
+		}
+		wig.Write(s.OutFile, records)
+	} else if s.AbsolutePercentError != "" {
+		second = wig.Read(s.AbsolutePercentError)
+		for i = range records {
+			chromIndex = getChromIndex(second, records[i].Chrom)
+			for j = range records[i].Values {
+				if records[i].Values[j] != 0 {
+					records[i].Values[j] = math.Abs((records[i].Values[j]-second[chromIndex].Values[j])/records[i].Values[j]) * 100
+				} else {
+					records[i].Values[j] = 0 //placeholder, these positions are undefined.
+				}
+			}
+		}
+		wig.Write(s.OutFile, records)
+	} else if s.Pearson != "" {
+		second = wig.Read(s.Pearson)
+		answer := wig.Pearson(records, second, s.Missing, s.SamplingFrequency)
+		out := fileio.EasyCreate(s.OutFile)
+		_, err = fmt.Fprintf(out, "PCC:\t%f\n", answer)
+		err = out.Close()
+		exception.PanicOnErr(err)
+	}
+}
+
+func multipleOptionCheck(s Settings) {
```

Comment:
Maybe this could have been coded with a case switch loop, where each option is a case and the default is log.Fatalf? But this is just a stylistic comment